provider/datagen/src/source.rs

```diff
@@ -104,20 +87,65 @@ impl SourceData {
 
     /// Adds ICU export data to this `DataSource`. The data will be downloaded from GitHub
     /// using the given tag. (see [GitHub releases](https://github.com/unicode-org/icu/releases)).
-    pub fn with_icuexport_for_tag(self, tag: &str) -> Result<Self, DataError> {
+    pub fn with_icuexport_for_tag(self, mut tag: &str) -> Result<Self, DataError> {
+        if tag == "release-71-1" {
+            tag = "icu4x/2022-06-30/71.x";
+        }
         self.with_icuexport(
             cached_path::CacheBuilder::new().freshness_lifetime(u64::MAX).build().and_then(|cache| cache
                 .cached_path(
                     &format!("https://github.com/unicode-org/icu/releases/download/{}/icuexportdata_{}.zip", tag, tag.replace('/', "-"))
             )).map_err(|e| DataError::custom("Download").with_display_context(&e))?)
     }
 
-    /// Creates a [`SourceData`] object with the latest data from GitHub.
-    pub fn latest(locale_subset: CldrLocaleSubset) -> Result<Self, DataError> {
-        // TODO query GitHub for the latest tags.
-        Self::default()
-            .with_cldr_for_tag("41.0.0", locale_subset)?
-            .with_icuexport_for_tag("release-71-1")
+    /// Adds CLDR data to this `DataSource`. This data will be downloaded from the `latest` GitHub tag.
```

Comment:
Thought: I think I prefer this to be an `async fn` but I don't feel strongly. It has to complete at the bginning of program execution anyway. Making it an `async fn` would mean that callers need to be in an async context, which may be a bigger change than we want/need to make right now.